### PR TITLE
Event registration system

### PR DESCRIPTION
**Pull request by rezural**
This adds the EventPlugin trait for registering events, and reworks the event registration code to use this within EditorPlugin.

### CODE REVIEW

File: src/plugin.rs [hunk 1]
```diff
@@ -77,8 +77,8 @@ enum EditorState {
     Editing,
 }
 
-fn add_editor_events(app: &mut AppBuilder) {
-    app.add_event::<edit_tools::events::TerraformerEvents>();
+fn register_editor_events(app: &mut AppBuilder) {
```
Comment (bonsairobo) [inline src/plugin.rs:80]: I would just inline this.

File: src/plugin.rs [hunk 2]
```diff
@@ -253,3 +253,7 @@ pub trait StatePlugin {
         set
     }
 }
+
+pub trait EventPlugin {
```
Comment (bonsairobo) [inline src/plugin.rs:257]: I'm not sure we actually need a trait for this. At least not yet.

**Review by bonsairobo — CHANGES_REQUESTED** [review]
Take a look at my comments. Nothing major, just style stuff.

File: src/plugin.rs (hunk 2 above)
Comment (rezural) [reply src/plugin.rs]: Fair enough, fixed this.

**Review by bonsairobo — APPROVED** [review]

**Pull request merged**